### PR TITLE
INT-1106  Contact groups in mobile and web UI

### PR DESCRIPTION
**Pull request by Onbehalfofme**

### CODE REVIEW

File: talk/talk-api.yaml [hunk 1]
```diff
@@ -3385,9 +3404,12 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
+          type: boolean
+          description: The parameter marks if this contact is favovurite for the user.
```
Comment (nurutdinov) [inline talk/talk-api.yaml:3409]:
```suggestion
          description: The parameter marks if this contact is favourite for the user.
```

File: talk/talk-api.yaml [hunk 2]
```diff
@@ -3334,6 +3344,15 @@ components:
...
```
Comment (nurutdinov) [inline talk/talk-api.yaml:3349]:
```suggestion
          description: The parameter marks if this contact is favourite for the user.
```

File: talk/talk-api.yaml [hunk 3]
```diff
@@ -3266,6 +3266,16 @@ components:
             fullName:
               description: The contact's first and last name.
               type: string
+            type:
+              type: string
+              description: The type of the contact.
+              enum:
```
Comment (nurutdinov) [inline talk/talk-api.yaml:3272]: Let's move it to a separate object

File: talk/talk-api.yaml [hunk 4]
```diff
@@ -3334,6 +3344,15 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
+          type: boolean
+          description: The parameter marks if this contact is favovurite for the user.
+        contactType:
+          type: string
+          description: The type of the contact.
+          enum:
```
Comment (nurutdinov) [inline talk/talk-api.yaml:3353]: Let's move it to a separate object

**Review by nurutdinov — CHANGES_REQUESTED** [review]

File: talk/talk-api.yaml [hunk 5]
```diff
@@ -3334,6 +3344,15 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
+          type: boolean
+          description: The parameter marks if this contact is favovurite for the user.
+        contactType:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:3350]: just `type`

File: talk/talk-api.yaml [hunk 6]
```diff
@@ -3334,6 +3344,15 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:3347]: required both?

File: talk/talk-api.yaml (hunk 5 above)
Comment (BullDog57Rus) [reply talk/talk-api.yaml:3345]: not fixed

**Review by BullDog57Rus — APPROVED** [review]

**Review by olyavertwist — APPROVED** [review]

File: talk/talk-api.yaml [hunk 7]
```diff
@@ -3413,13 +3415,19 @@ components:
...
         - PERSONAL
+        - INTERNAL
         - OTHER
     ContactPhoneNumber:
       type: object
       description: The object contains the contact's phone number details.
       required:
         - phoneNumber
         - type
+      discriminator:
```
Comment (BullDog57Rus) [inline talk/talk-api.yaml:3426]: I think moving discriminator on the contact level will also solve the problem of statuses for internal contacts

**Comment by olyavertwist** [comment]
`
favourite:
              type: boolean
              description: The parameter marks if this contact is favourite for the user.
`

Hey, I'm wondering what's better 'favorite' or 'favourite'? Both spellings are correct but the one without U (favorite) is more common in the US. At the same time, Apple prefers 'favourite' while Microsoft says 'favorite'. 

In any case, we should use the same spelling here in API and in UI. @nurutdinov What do you plan on using in UI? In Confluence article. I see 'favorites' and would personally prefer it too but it's not really a bit deal. (https://blackbirdsoftware.atlassian.net/wiki/spaces/IN/pages/1985871972/Contacts+List)

File: talk/talk-api.yaml [hunk 8]
```diff
@@ -3406,6 +3436,24 @@ components:
...
+    InternalContactPhoneNumber:
+      type: object
+      description: The request model to log an activity for a lead that subscribed to the Inperium mail list.
+      allOf:
+        - $ref: "#/components/schemas/ContactPhoneNumber"
+        - type: object
+          properties:
+            name:
+              type: string
+              description: The name for the phone number.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:3448]:
```suggestion
              description: The phone number title.
```

File: talk/talk-api.yaml [hunk 9]
```diff
@@ -3406,6 +3436,24 @@ components:
...
+            countryCode:
+              $ref: "#/components/schemas/CountryCode"
+            city:
+              type: string
+              description: The city where phone number resides.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:3453]:
```suggestion
              description: The city where the phone number resides.
```

File: talk/talk-api.yaml [hunk 10]
```diff
@@ -3406,6 +3436,24 @@ components:
...
+            state:
+              type: string
+              description: The state where phone number resides.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:3456]:
```suggestion
              description: The state where the phone number resides.
```

File: talk/talk-api.yaml [hunk 11]
```diff
@@ -3773,18 +3821,20 @@ components:
...
-            - IN_A_CALL
-            - OFFLINE
+          $ref: "#/components/schemas/SystemStatus"
         until:
           type: integer
           description: This parameter defines when to reset a status, e.g., display user status as DO_NOT_DISTURB until 12 pm.
           format: int64
+    SystemStatus:
+      type: string
+      description: The user state in the system. The status is displayed to other users. This is enum parameter.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:3831]:
```suggestion
      description: The user's state in the system. The status is displayed to other users. This is enum parameter.
```

File: talk/talk-api.yaml [hunk 12]
```diff
@@ -3385,19 +3406,28 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
+          type: boolean
+          description: The parameter marks if this contact is favourite for the user.
     ContactInfoType:
       type: string
-      description: The type of contact.
+      description: The type of contact info.
```
Comment (olyavertwist) [inline talk/talk-api.yaml:3414]:
```suggestion
      description: The type of contact info. This is enum parameter.
```

**Review by olyavertwist — CHANGES_REQUESTED** [review]

**Comment by nurutdinov** [comment]
>

**Comment by nurutdinov** [comment]
@olyavertwist, yes, Evgeni used 'favorite' in the design too. @Onbehalfofme let's use it everywhere.

File: talk/talk-api.yaml [hunk 13]
```diff
@@ -3334,6 +3341,19 @@ components:
           description: The list of emails.
           items:
             $ref: "#/components/schemas/ContactEmail"
+        favourite:
+          type: boolean
+          description: The parameter marks if this contact is favourite for the user.
+        type:
+          $ref: "#/components/schemas/ContactType"
```
[the remark this answers is not in the document]
Comment (Onbehalfofme) [reply talk/talk-api.yaml:3348]: User status contains time which has to be available to the user itself.

**Review by nurutdinov — APPROVED** [review]

**Review by olyavertwist — APPROVED** [review]

**Pull request merged**